Include/internal/pycore_optimizer.h

```diff
@@ -193,27 +184,72 @@ static inline uint16_t uop_get_error_target(const _PyUOpInstruction *inst)
 // handle before rejoining the rest of the program.
 #define MAX_CHAIN_DEPTH 4
 
-typedef struct _Py_UopsSymbol _Py_UopsSymbol;
+/* Symbols */
+/* See explanation in optimizer_symbols.c */
+
+
+typedef enum _JitSymType {
+    JIT_SYM_UNKNOWN_TAG = 0,
+
+    JIT_SYM_NULL_TAG = 2,
+    JIT_SYM_NON_NULL_TAG = 3,
+    JIT_SYM_BOTTOM_TAG = 4,
+    JIT_SYM_TYPE_VERSION_TAG = 5,
+    JIT_SYM_KNOWN_CLASS_TAG = 6,
+    JIT_SYM_KNOWN_VALUE_TAG = 7,
+    JIT_SYM_TUPLE_TAG = 8,
+} JitSymType;
+
+typedef struct _jit_opt_known_class {
+    uint8_t tag;
+    uint32_t version;
+    PyTypeObject *type;
+} JitOptKnownClass;
```

Comment:
We could probably just get the version by doing `cls.type->tp_version_tag`, right? Having both here just creates opportunities for them to be stale or out-of-sync, I think.

Reply:
`sym->cls.type->tp_version_tag` is the version tag of the value's class now. `sym->version.version` is the version tag it would have should execution reach this point in the code. Not quite the same thing.

Reply:
I would imagine that a `sym->cls.version` that's out-of-sync with "sym->cls.type->tp_version_tag` should result in the bottom type, then, right? It's not possible in future code, since the cached tag is stale.